src/sagas/helpers.js

```diff
@@ -275,3 +279,54 @@ export function getNetworkSettings(state) {
   // has precedence, once it indicates a custom network.
   return STORE.getItem(networkSettingsKeyMap.networkSettings) ?? state.networkSettings;
 }
+
+/**
+ * A request abstraction that applies a progressive retry strategy.
+ * One can define how many retries it should make or use the default value.
+ *
+ * @param {Promise<any>} request The async callback function to be executed.
+ * @param {number} maxRetries The max retries allowed, with default value.
+ * Notice this param should be at least 2 to make sense.
+ * @returns {{ success?: any; error?: any } A success object from the request,
+ * or an error definition after retries exhausted.
+ *
+ * @example
+ * yield call(progressiveRetryRequest, async () => asyncFn());
+ * // use default maxRetries
+ *
+ * @example
+ * yield call(progressiveRetryRequest, async () => asyncFn(), 3);
+ * // use custom maxRetries equal to 3
+ */
+export function* progressiveRetryRequest(request, maxRetries = MAX_RETRIES_WS_CALL) {
+  const resultHandler = async (cb) => {
+    try {
+      const success = await cb();
+      return { success };
+    } catch (error) {
+      return { error }
+    }
+  };
+
+  let result = null;
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < maxRetries; i++) {
+    result = yield call(resultHandler, request);
+    if (result.success) {
+      return result;
+    }
+
+    // skip delay for last call
+    if (i === maxRetries - 1) {
+      continue;
+    }
+
+    // attempt 0: 300ms
+    // attempt 1: 330ms
+    // attempt 2: 420ms
+    // attempt 3: 570ms
+    // attempt 4: 780ms
```

Comment:
You are trying 5 times in less than 2.5s. If the user had a network instability, this might not be enough for him to recover. I understand that we can't try forever and it doesn't matter the value we choose, it might not be enough anyway for the network to recover. But I think we could increase this amount, maybe even increase 1s in each attempt, then you give the user network 5s to recover (1s - 1s - 1s - 1s - 1s). 

What do you think?

Reply:
Well, my experience was that these network flash instabilities didn't take long, actually. I believe 2.5s is enough for now imo.

Reply:
Changed to 8 attempts by default, the effect almost the same. Done at https://github.com/HathorNetwork/hathor-wallet-mobile/pull/444/commits/c552aa618f7da09b9353666901f6bfe4d6ca441e